src/waypoint.jsx

```diff
@@ -333,13 +352,20 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
-    // We need an element that we can locate in the DOM to determine where it is
-    // rendered relative to the top of its context.
-    return <span ref={this.refElement} style={{ fontSize: 0 }} />;
+    if (this.props.children) {
+      // TODO: consider returning children directly when there is only one child,
+      // using React.cloneElement to attach a ref (but child might already have a ref)
```

Comment:
It's also a tad unfortunate to have to inject an element which may interfere with syling. But I can remove the TODO because it's unlikely to be worth it